app/src/pages/inside/settingsPage/settingTabs/autoAnalysisTab/analysisForm/analysisForm.jsx

```diff
@@ -0,0 +1,231 @@
+    initialize: PropTypes.func,
+    change: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    showNotification: PropTypes.func,
+    updateAutoAnalysisConfigurationAction: PropTypes.func,
+    formInputsValues: PropTypes.object,
+  };
+
+  static defaultProps = {
+    analyzerConfiguration: {},
+    currentProject: '',
+    initialize: () => {},
+    change: () => {},
+    handleSubmit: () => {},
+    showNotification: () => {},
+    updateAutoAnalysisConfigurationAction: () => {},
+    formInputsValues: {},
+  };
+
+  state = {
+    autoAnalysisMode: DEFAULT_ANALYSIS_MODE,
+  };
+
+  componentDidMount() {
+    const initialConfiguration = {
+      ...this.props.analyzerConfiguration,
+    };
+    delete initialConfiguration.indexing_running;
+    this.setAnalysisMode(initialConfiguration);
+    this.props.initialize(initialConfiguration);
+  }
+
+  onInputChange = (event, newValue, previousValue, name) => {
+    const formInputsValues = this.props.formInputsValues;
+    formInputsValues[name] = newValue;
+    this.setAnalysisMode(formInputsValues);
+  };
+
+  onFormSubmit = (data) => {
+    const dataToSend = {
+      configuration: {
+        analyzerConfiguration: {
+          ...data,
+        },
+      },
+    };
+    fetch(URLS.project(this.props.currentProject), { method: 'put', data: dataToSend })
+      .then(() => {
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(messages.updateSuccessNotification),
+          type: NOTIFICATION_TYPES.SUCCESS,
+        });
+        this.props.updateAutoAnalysisConfigurationAction(dataToSend);
+      })
+      .catch(() => {
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(messages.updateErrorNotification),
+          type: NOTIFICATION_TYPES.ERROR,
+        });
+      });
+  };
+
+  setAnalysisMode = (modeConfig) => {
+    const analysisModeKeys = Object.keys(analysisModeConfig);
+    const existingMode = analysisModeKeys.find(
+      (key) =>
+        analysisModeConfig[key].minShouldMatch === Number(modeConfig.minShouldMatch) &&
+        analysisModeConfig[key].minDocFreq === Number(modeConfig.minDocFreq) &&
+        analysisModeConfig[key].minTermFreq === Number(modeConfig.minTermFreq) &&
+        analysisModeConfig[key].numberOfLogLines === Number(modeConfig.numberOfLogLines),
+    );
+    this.setState({
+      autoAnalysisMode: existingMode || '',
+    });
+  };
+
+  tabChangeHandle = (newValue) => {
+    this.props.change('minShouldMatch', analysisModeConfig[newValue].minShouldMatch);
+    this.props.change('minDocFreq', analysisModeConfig[newValue].minDocFreq);
+    this.props.change('minTermFreq', analysisModeConfig[newValue].minTermFreq);
+    this.props.change('numberOfLogLines', analysisModeConfig[newValue].numberOfLogLines);
+
+    this.setState({
+      autoAnalysisMode: newValue,
+    });
+  };
+
+  render() {
+    const { intl, handleSubmit } = this.props;
+
+    const tabItems = [
```

Comment:
this is a static data, so, please, move it to the class filed. Because in that case, with each rerender will be created a new array.

Reply:
Fixed